artifactory/commands/repository/repository.go

```diff
@@ -36,51 +39,133 @@ func (rc *RepoCommand) TemplatePath() string {
+		var handlerFunc func(servicesManager artifactory.ArtifactoryServicesManager, jsonConfig []byte, isUpdate bool) error
+		packageType := fmt.Sprint(repoConfigMap[PackageType])
+		switch repoConfigMap[Rclass] {
+		case Local:
+			handlerFunc = localRepoHandlers[packageType]
+		case Remote:
+			handlerFunc = remoteRepoHandlers[packageType]
+		case Virtual:
+			handlerFunc = virtualRepoHandlers[packageType]
+		case Federated:
+			handlerFunc = federatedRepoHandlers[packageType]
+		default:
+			return errorutils.CheckErrorf("unsupported rclass: %s", repoConfigMap[Rclass])
+		}
+		if handlerFunc == nil {
+			return errors.New("unsupported package type: " + packageType)
+		}
+
+		if err := handlerFunc(servicesManager, content, isUpdate); err != nil {
+			return err
 		}
 	}
-	// Write a JSON with the correct values
-	content, err := json.Marshal(repoConfigMap)
+	return nil
+}
+
+func multipleRepoHandler(servicesManager artifactory.ArtifactoryServicesManager, jsonConfig []byte, isUpdate bool) (err error) {
+	artifactoryVersion, err := servicesManager.GetVersion()
 	if err != nil {
-		return err
+		return errorutils.CheckErrorf("failed to get Artifactory rtVersion: %s", err.Error())
+	}
+	rtVersion := version.NewVersion(artifactoryVersion)
+	if isUpdate {
+		if !rtVersion.AtLeast("7.104.2") {
+			return errorutils.CheckErrorf("bulk repository updation is supported from Artifactory rtVersion 7.104.2, current rtVersion: %v", artifactoryVersion)
+		}
+	} else {
+		if !rtVersion.AtLeast("7.84.3") {
+			return errorutils.CheckErrorf("bulk repository creation is supported from Artifactory rtVersion 7.84.3, current rtVersion: %v", artifactoryVersion)
+		}
 	}
 
-	servicesManager, err := rtUtils.CreateServiceManager(rc.serverDetails, -1, 0, false)
+	log.Info("creating/updating repositories in batch...")
+
+	err = servicesManager.CreateUpdateRepositoriesInBatch(jsonConfig, isUpdate)
 	if err != nil {
 		return err
 	}
-	// Rclass and packageType are mandatory keys in our templates
-	// Using their values we'll pick the suitable handler from one of the handler maps to create/update a repository
-	var handlerFunc func(servicesManager artifactory.ArtifactoryServicesManager, jsonConfig []byte, isUpdate bool) error
-	packageType := fmt.Sprint(repoConfigMap[PackageType])
-	switch repoConfigMap[Rclass] {
-	case Local:
-		handlerFunc = localRepoHandlers[packageType]
-	case Remote:
-		handlerFunc = remoteRepoHandlers[packageType]
-	case Virtual:
-		handlerFunc = virtualRepoHandlers[packageType]
-	case Federated:
-		handlerFunc = federatedRepoHandlers[packageType]
-	default:
-		return errorutils.CheckErrorf("unsupported rclass: %s", repoConfigMap[Rclass])
-	}
-	if handlerFunc == nil {
-		return errors.New("unsupported package type: " + packageType)
-	}
-	return handlerFunc(servicesManager, content, isUpdate)
+
+	if isUpdate {
+		log.Info("Successfully updated the repositories")
+	} else {
+		log.Info("Successfully created the repositories")
+	}
```

Comment:
Based on the command user should be clear here whether it is update or create so having a same log message should be ok.

```suggestion
	log.Info("Successfully created/updated the repositories")
```